### PR TITLE
endsWith added.

### PR DESCRIPTION
**Pull request by mfirry**
This is really just @gcanti being my puppeteer :)

### CODE REVIEW

File: src/endsWith.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+export function endsWith<A>(
+  E: Eq<A>
+): {
+  (suffix: Array<A>): (as: Array<A>) => boolean;
+  (suffix: Array<A>, as: Array<A>): boolean;
+};
+
+export function endsWith<A>(E: Eq<A>): (suffix: Array<A>, oas?: Array<A>) => any {
```
Comment (giogonzo) [inline src/endsWith.ts]: minor: `oas` as a name was temporarily introduced in previous PRs to fight against the `no-shadowed-variable` TSLint rule. I think that sticking to `as` here is clearer.

**Comment by mfirry** [comment]
Not sure I'm fully able to do both perfs and dtslint. I'll give it a look and let you know.

File: src/endsWith.ts (hunk 1 above)
Comment (mfirry) [reply src/endsWith.ts]: You're right. I was going for `as` and `bs` then switched to `suffix` and `list` (used in ramda code). Then @gcanti took it over.

**Comment by giogonzo** [comment]
thanks @mfirry

**Pull request merged**